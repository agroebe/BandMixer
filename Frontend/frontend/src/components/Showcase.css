.jumbo {
    /* background-image: url("https://www.designbolts.com/wp-content/uploads/2012/12/simple-pattern-white-seamless-website-background.jpg");
    background-size: cover; */
}