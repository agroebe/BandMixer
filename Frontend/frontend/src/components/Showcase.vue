<template>
    <b-jumbotron header="BandMixer" lead="Let us help you find your jam!">
        <p>Start your search below</p>
        <div class="search">
            <b-form class="d-inline" inline>
                <b-form-select class="mr-2" id="input" v-model="form.instrument" :options="instruments" required></b-form-select>
                <b-form-input class="mr-2" v-model="form.location" placeholder="ZIP code or location" required></b-form-input>
                <b-button variant="primary" @click="submit()">Submit</b-button>
            </b-form>
        </div>
    </b-jumbotron>
</template>

<script>
export default {
    name: 'Showcase',
    data() {
        return {
            form: {
                instrument: null,
                location: ''
            },
            instruments: [ { text: 'What are you looking for?', value: null }, 'All instruments', 'Guitar', 'Bass', 'Drums', 'Piano', 'Keyboard' ]
        }
    },
    methods: {
        submit() {
            // evt.preventDefault()
            // alert(JSON.stringify(this.form))
            this.$router.push({
                name: 'Results',
                params: {
                    test: 'hey there!'
                }
            })
        },
    }
}
</script>