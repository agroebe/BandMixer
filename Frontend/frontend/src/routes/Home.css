hr {
    border: none;
    height: 50px;
    border-bottom: 1px solid #4d4c4b;
    box-shadow: 0 10px 20px -20px #333;
    margin: -25px auto 10px; 
  }