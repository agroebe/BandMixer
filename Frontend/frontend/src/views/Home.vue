<template>
    <b-card
        bg-variant="dark"
        text-variant="white"
        title="Learn About BandMixer"
        img-src="https://images.unsplash.com/photo-1510915361894-db8b60106cb1?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1350&q=80"
        img-width="300"
        img-alt="Card image"
        img-left
        class="m-auto"
        style="max-width: 65rem;"
        >
        <b-card-text>BandMixer helps connect music enthusiasts of varying competencies, skill levels, and genres.</b-card-text>
        <b-card-text>BandMixer helps connect music enthusiasts.</b-card-text>
    </b-card>
</template>