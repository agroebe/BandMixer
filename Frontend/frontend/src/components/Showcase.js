import React from 'react';
import { withRouter } from 'react-router-dom'
import { Link } from 'react-router-dom';
import { Jumbotron, Form, Button, Modal } from 'react-bootstrap';
import './Showcase.css'
import { MusicNote } from 'react-bootstrap-icons'

export default class Showcase extends React.Component {
    state = {
        instrumentOptions: [
            'Guitar', 'Bass', 'Drums', 'Piano', 'Keyboard'
        ],
        locationOptions: [
            'California', 'Florida', 'Georgia', 'Illinois', 'Iowa', 'Michigan', 'Minnesota', 'New-Jersey', 'New-York', 'Pennsylvania', 'Texas', 'Washington'
        ],
        instrument: '',
        location: '',
        showBeginnerTutorial: false
    }


    search() {
    }

    render() {
        return(
            <>
                <Modal size="xl" show={ this.state.showBeginnerTutorial } onHide={ () => this.setState({ showBeginnerTutorial: false }) }>
                    <Modal.Header>
                        <Modal.Title>BandMixer - Beginner Tutorial</Modal.Title>
                    </Modal.Header>

                    <Modal.Body>
                        <h5><strong>Creating an account</strong></h5>
                        <p>To create an account with BandMixer, locate the buttons in the top right of this website's navigation bar. After doing so, you will want to click the "Register"
                            Button. Fill out the required fields, ensuring that you will remember what you enter in as your username and password. <strong>Note: registering an account does
                            NOT create your musician profile. You will do so after registering and signing-in.</strong></p>

                        <br/>

                        <h5><strong>Creating your profile</strong></h5>
                        <p>As mentioned above, your musician profile will not be created upon registration; you will need to follow some more steps to create one. Once signed in, navigate
                            to your "Profile" button in the top right of the navigation bar. This should open a dialogue box that allows you to enter some basic information to your musician profile,
                            such as instruments, skill levels, genre preferences, etc,. From this form, you will also be able to set your privacy options and determine whether or not you want to 
                            advertise your musician profile to the public. If you choose to advertise your profile, it will appear in search terms of other users, and you may be contacted by
                            other musicians seeking bandmates!</p>

                        <br/>

                        <h5><strong>Start your search</strong></h5>
                        <p>To take full advantage of BandMixer, you will need to know how to navigate our search functionality. The basic search seen on the homepage should be extremely
                            self-explanatory; simply enter an instrument and location, and we will do the rest for you! After doing so, you should be able to see any results we could match 
                            with your search terms. From that point, you can feel free to reach out to any musicians you are interested in by looking at their contact information. 
                            An important note is that <strong>you do not need an account to search for musician profiles</strong>.
                        </p>


                    </Modal.Body>

                    <Modal.Footer>
                        <Button variant="danger" onClick={ () => this.setState({ showBeginnerTutorial: false }) }>Cancel</Button>
                    </Modal.Footer>
                </Modal>
                <Jumbotron className="text-center jumbo">
                    <h1 className="display-2">BandMixer</h1>
                    <p className="lead"><MusicNote/>Let's help you find your jam - start your search below!</p>
                    <Form inline className="justify-content-center mb-3" onSubmit={ this.search }>
                        <Form.Control as="select" className="mr-2" onChange={ e => this.setState({ instrument: e.target.value }) } >
                            <option value={0}>Select what you're looking for...</option>
                            { this.state.instrumentOptions.map((value, index) => {
                                return <option value={ index + 1} key={ index }>{ value }</option>
                            }) }
                        </Form.Control>
                        <Form.Control as="select" className="mr-2" onChange={ e => this.setState({ location: e.target.value }) } >
                            <option value={0}>Select a location...</option>
                            { this.state.locationOptions.map((value, index) => {
                                return <option value={ index + 1} key={ index }>{ value }</option>
                            }) }
                        </Form.Control>
                        { this.state.location && this.state.instrument ? (
                            <Link to={{
                                pathname: '/results',
                                state: {
                                    location: this.state.location,
                                    instrument: this.state.instrument
                                }
                            }}><Button>Search</Button></Link>
                        ) : (
                            <Button onClick={ () => alert('Please double check your search terms, ensuring both and instrument and location are selected.')}>Search</Button>
                        )}
                    </Form>
                    <h5 className="d-inline">Confused? Try out <a onClick={ () => this.setState({ showBeginnerTutorial: true }) } className="link">our beginner tutorial</a>!</h5>
                </Jumbotron>
            </>
        )
    }
}

withRouter(Showcase)