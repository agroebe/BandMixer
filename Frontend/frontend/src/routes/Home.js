import React from 'react';
import Showcase from '../components/Showcase'
import axios from 'axios'
import './Home.css'
import { CardDeck, Card, Button } from 'react-bootstrap';

export default class Home extends React.Component {
    state = {
        posts: []
    }

    componentDidMount() {
        axios.get('http://coms-309-cy-01.cs.iastate.edu:8080/posts/all').then(r => {
            r.data.forEach(post => {
                var joined = this.state.posts.concat(post)
                this.setState( { posts: joined })
            })
        })
    }

    render() {
        return(
            <>
                <Showcase loc={this.props.loc}/>
                <CardDeck className="mx-auto w-75 pb-10">
                <Card>
                    <Card.Img variant="top" src="https://www.superprof.co.uk/blog/wp-content/uploads/2017/07/guitar-guitar-player-musician-instrument-guitarist-1060x707.jpg" height="275px"/>
                    <Card.Body>
                        <Card.Title>Create your musician profile</Card.Title>
                        <Card.Text>
                            Get started right away with a fully-customizable musician profile, allowing you to showcase
                            your instrument competencies, skill level, and genre and location preferences.
                        </Card.Text>
                        <Card.Text>
                            After doing so, we'll use the information you provide to help you find bandmates through our advanced searching.
                        </Card.Text>
                    </Card.Body>
                    <Card.Footer className="text-center">
                        <Button>Get started</Button>
                    </Card.Footer>
                </Card>
                <Card>
                    <Card.Img variant="top" src="https://www.bmi.com/cache/images/news/2019/iStock-1125877063_770_514_70_s.jpg" height="275px"/>
                    <Card.Body>
                        <Card.Title>Start your search for bandmates</Card.Title>
                        <Card.Text>
                            Finding bandmates won't be so hard anymore! We give users the power to create bands, join existing ones, or even
                            just find some musician buddies to learn from and jam with.
                        </Card.Text>
                        <Card.Text>
                            You'll be able to search for other musicians by instrument, skill levels, genre preferences, and most importantly, location!
                        </Card.Text>
                    </Card.Body>
                    <Card.Footer className="text-center">
                        <Button href="#">Start searching</Button>
                    </Card.Footer>
                </Card>
                <Card>
                    <Card.Img variant="top" src="https://mentalitch.com/wp-content/uploads/2019/07/Top-British-Bands-of-the-80s.jpeg" height="275px"/>
                    <Card.Body>
                        <Card.Title>Start your band</Card.Title>
                        <Card.Text>
                            BandMixer connects musicians with the ultimate goal of enabling the formation of bands and music groups.
                        </Card.Text>
                        <Card.Text>
                            No matter the instrument, skill level, or music taste, we want to allow musicians to live out their dreams of
                            being in a band.
                        </Card.Text>
                    </Card.Body>
                    <Card.Footer className="text-center">
                        <Button>Success stories</Button>
                    </Card.Footer>
                </Card>
                </CardDeck>

                <br></br><hr className="w-75" style={{ color: "black"}}></hr><br></br>

                <h1 className="text-center">Featured posts:</h1>
                <p className="text-center">Click <a href="/explore">here</a> to see more.</p>
                <CardDeck className="mx-auto w-75">
                { this.state.posts.slice(0, 5).map(post => (
                    <Card style={{  marginLeft: '5px', marginRight: '5px', marginBottom: '10px', marginTop: '10px' }}>
                        <Card.Body>
                        <Card.Title>{ post.title }</Card.Title>
                        <Card.Subtitle className="mb-2 text-muted">Posted by { post.owner.username }</Card.Subtitle>
                        <Card.Text>
                            { post.textContent }
                        </Card.Text>
                        <Card.Link href={'/post/' + post.id }>View Post</Card.Link>
                        </Card.Body>
                    </Card>
                ))}
                </CardDeck>

                <br></br><hr className="w-75"></hr><br></br>
                <br></br><br></br>
            </>

        )
    }
}