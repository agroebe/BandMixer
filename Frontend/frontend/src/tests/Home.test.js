import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Home from '../routes/Home'
import Showcase from '../components/Showcase'
import { CardDeck } from 'react-bootstrap';

it('renders without crashing', () => {
    const home = renderer.create(
        <BrowserRouter><Home/></BrowserRouter> // Wrap test in BrowserRouter so that Link and Route properly works
    )

    let tree = home.toJSON()
    expect(Object.keys(tree).length).not.toEqual(0)
})

it('properly renders subcomponents', () => {
    const home = renderer.create(
        <BrowserRouter><Home/></BrowserRouter> // Wrap test in BrowserRouter so that Link and Route properly works
    )

    const homeInstance = home.root

    expect(homeInstance.findByType(Showcase)).not.toEqual(null) // Showcase is a sub-component of home
    expect(homeInstance.findAllByType(CardDeck)).not.toEqual(null)
})