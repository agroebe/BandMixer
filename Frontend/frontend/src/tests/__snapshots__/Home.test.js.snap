// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <div
    className="text-center jumbo jumbotron"
  >
    <h1
      className="display-2"
    >
      BandMixer
    </h1>
    <p
      className="lead"
    >
      Let's help you find your jam - start your search below!
    </p>
    <form
      className="justify-content-center mb-3 form-inline"
      onSubmit={[Function]}
    >
      <select
        className="mr-2 form-control"
        onChange={[Function]}
      >
        <option
          selected={true}
          value={0}
        >
          Select what you're looking for
        </option>
        <option
          value={1}
        >
          Guitar
        </option>
        <option
          value={2}
        >
          Bass
        </option>
        <option
          value={3}
        >
          Drums
        </option>
        <option
          value={4}
        >
          Piano
        </option>
        <option
          value={5}
        >
          Keyboard
        </option>
      </select>
      <input
        className="mr-2 form-control"
        onChange={[Function]}
        placeholder="ZIP code or location"
        type="text"
      />
    </form>
    <h5
      className="d-inline"
    >
      Confused? Try out 
      <a
        href="#"
        onClick={[Function]}
      >
        our beginner tutorial
      </a>
      !
    </h5>
  </div>,
  <div
    className="mx-auto w-75 card-deck"
  >
    <div
      className="card"
    >
      <img
        className="card-img-top"
        height="275px"
        src="https://www.superprof.co.uk/blog/wp-content/uploads/2017/07/guitar-guitar-player-musician-instrument-guitarist-1060x707.jpg"
      />
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Create your musician profile
        </div>
        <p
          className="card-text"
        >
          Get started right away with a fully-customizable musician profile, allowing you to showcase your instrument competencies, skill level, and genre and location preferences.
        </p>
        <p
          className="card-text"
        >
          After doing so, we'll use the information you provide to help you find bandmates through our advanced searching.
        </p>
      </div>
      <div
        className="text-center card-footer"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          Get started
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        className="card-img-top"
        height="275px"
        src="https://www.bmi.com/cache/images/news/2019/iStock-1125877063_770_514_70_s.jpg"
      />
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Start your search for bandmates
        </div>
        <p
          className="card-text"
        >
          Finding bandmates won't be so hard anymore! We give users the power to create bands, join existing ones, or even just find some musician buddies to learn from and jam with.
        </p>
        <p
          className="card-text"
        >
          You'll be able to search for other musicians by instrument, skill levels, genre preferences, and most importantly, location!
        </p>
      </div>
      <div
        className="text-center card-footer"
      >
        <a
          className="btn btn-primary"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Start searching
        </a>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        className="card-img-top"
        height="275px"
        src="https://mentalitch.com/wp-content/uploads/2019/07/Top-British-Bands-of-the-80s.jpeg"
      />
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Start your band
        </div>
        <p
          className="card-text"
        >
          BandMixer connects musicians with the ultimate goal of enabling the formation of bands and music groups.
        </p>
        <p
          className="card-text"
        >
          No matter the instrument, skill level, or music taste, we want to allow musicians to live out their dreams of being in a band.
        </p>
      </div>
      <div
        className="text-center card-footer"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          Success stories
        </button>
      </div>
    </div>
  </div>,
]
`;
